src/main.rs

```diff
@@ -20,9 +25,97 @@ enum Opt {
         /// Verbose mode (-v, -vv, -vvv, etc.)
         #[clap(short, long, parse(from_occurrences))]
         verbose: u8,
+
+        #[clap(short, long)]
+        google_sheet: Option<String>,
     },
 }
 
+struct Collection<T: google_sheets::IntoSheetEntry> {
```

Comment:
There's a lot here. At minimum we need another module, but also some comments to explain the intention. I'm going to take a bit of time to read into it but I may hit my quota and have to come back to it.